lib/src/user_location_layer.dart

```diff
@@ -66,14 +68,10 @@ class _MapsPluginLayerState extends State<MapsPluginLayer>
       switch (state) {
         case AppLifecycleState.inactive:
         case AppLifecycleState.paused:
-          if (_onLocationChangedStreamSubscription != null) {
-            _onLocationChangedStreamSubscription.cancel();
-          }
+          _cancel(_onLocationChangedStreamSubscription);
           break;
         case AppLifecycleState.resumed:
-          if (_onLocationChangedStreamSubscription != null) {
-            _onLocationChangedStreamSubscription.resume();
-          }
+          _cancel(_onLocationChangedStreamSubscription);
```

Comment:
Although this works, why are we closing the stream in both the cases? Just curious

Reply:
Oh, sorry I must have missed that, my bad

Reply:
I am sorry for the trouble, but could you send a PR to fix it,  if possible?

Reply:
No problem, I don't have access to my pc right now, but I'll see if I can do it tomorrow

Reply:
Thanks, my laptop is kinda broke so...running OS from the live-image as of now